dlt/dlt_broker_handlers.py

```diff
@@ -199,6 +200,126 @@ def stop(self):
+            else:
+                queue_id, filters, add = msg
+
+            if add:
+                for apid_ctid in filters:
+                    self.context_map[apid_ctid].append(queue_id)
+            else:
+                try:
+                    for apid_ctid in filters:
+                        self.context_map[apid_ctid].remove(queue_id)
+                        if not self.context_map[apid_ctid]:
+                            del self.context_map[apid_ctid]
+                except (KeyError, ValueError):
+                    # - queue_id already removed or not inserted
+                    pass
+
+            if self.filter_ack_queue and queue_ack_id:
+                logger.debug("Send filter ack message: queue_ack_id: %s, add: %s", queue_ack_id, add)
+                self.filter_ack_queue.put((queue_ack_id, add))
+
+    def handle(self, message):
+        """Function to be called for every message received
+
+        :param DLTMessage message: received new DLTMessage instance
+        :returns: True if the loop should continue, False to stop the loop and exit
+        :rtype: bool
+        """
+        self._process_filter_queue()
+
+        if message:
+            # Dispatch the message
+            msg_ctx = ((message.apid, message.ctid), (None, None), (message.apid, None), (None, message.ctid))
+            qids = (
+                queue_id
+                for filters, queue_ids in self.context_map.items()
+                for queue_id in queue_ids
+                if filters in msg_ctx
+            )
+            for queue_id in qids:
+                if self.message_queue.full():
+                    logger.error("message_queue is full ! put() on this queue will block")
+                self.message_queue.put((queue_id, message))
+                logger.debug("Add to message_queue")
+
+            # Send the message's timestamp
+            if self._dlt_time_value:
+                self._dlt_time_value.timestamp = message.storage_timestamp
+
+        return not self.mp_stop_flag.is_set()
+
+    def run(self):
+        """DLTMessageHandler worker method"""
+        logger.info("Start to process dlt file %s", self.file_name)
+
+        while not self.mp_stop_flag.is_set():
+            # If dlt file does not exist, then wait until it exists
+            if not os.path.exists(self.file_name):
+                time.sleep(0.1)
+                # File not-existing only happens at the very beginning of dlt broker
+                # Users might call dlt_broker.add_context() at this tricky moment
+                # So, we should not block the handling of context by calling handle() with None
+                self.handle(None)
+                continue
+            elif not self.dlt_reader:
+                # If dlt file exists, but no reader is instantiated, then instantiate
+                self.dlt_reader = cDLTFile(filename=self.file_name, is_live=True, iterate_unblock_mode=True)
```

Comment:
I will not review the entire DLTFilespinner since it needs to go away and is not needed. But this part will be relevant in the DLTMessagehandler for files, hence I am commenting here.

Please use the load method correctly with corresponding filters from dlt module to do this, instead of directly calling cDLTFile.

Reply:
why cannot I use cDLTFile directly? will it hurt me in anyway? if it does not bring any extra benefits of using load(), then it is my decision to use which way I would prefer.

Reply:
Please review DLTFileSpinner. If you reject to do that, then you cannot get the whole picture.

DLTFileSpinner is another use case of DLTMessageHandler, for such absolute truth, i knew that from the very beginning. And that is also why Wenshi does not like my way of copy-and-paste the majority part of DLTMessageHandler code, in order to achieve "not touching the branch of DLTConnector". I dont mean the ugly state is the end state, I'm aware of what I'm doing, what I have in mind is to split the big and complicated task into two smaller steps:

simple append a new functionality for IOC to mtee, without touching DLTConnector from DLTClient
code refactory and bridge DLTFileSpinner and DLTMessageHandler together. At this step, only python-dlt needs to be worked on, i dont even need to touch mtee_core, mtee_gen22, and idcevo. Because the used API will stay the same there in the uplevel.
what I want to achieve with such split: in case somebody complains DLTConnector does not work after this big change, then I dont need to revoke all 2 steps, right? because step 1 does not touch DLTConnector with DLTClient at all. I just need to focus on the change made in step2, and the worst case is to revoke step 2 from python-dlt without change mtee_core, mtee_gen22, idcevo, because the added IOC functionality introduced from step 1 stays there.

Reply:
@lzhou3957 Please remove your comment. There are a lot of internal repo names and projects. Please consider the confidential rules...